sdk/textanalytics/Azure.AI.TextAnalytics/src/HealthcareEntity.cs

```diff
@@ -1,20 +1,144 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
+using System;
 using System.Collections.Generic;
-using Azure.Core;
+using System.Globalization;
+using System.Linq;
+using System.Text.RegularExpressions;
+using Azure.AI.TextAnalytics.Models;
 
 namespace Azure.AI.TextAnalytics
 {
     /// <summary>
-    /// HealthcareEntity.
+    /// .
     /// </summary>
-    [CodeGenModel("HealthcareEntity")]
-    public partial class HealthcareEntity
+    public class HealthcareEntity
     {
+        internal HealthcareEntity(HealthcareEntityInternal entity, IEnumerable<HealthcareEntityInternal> healthcareEntities = default, IEnumerable<HealthcareRelationInternal> healthcareRelations = default)
+        {
+            Category = entity.Category;
+            Text = entity.Text;
+            SubCategory = entity.Subcategory;
+            ConfidenceScore = entity.ConfidenceScore;
+            Offset = entity.Offset;
+            DataSources = entity.Links;
+            if (healthcareEntities != null && healthcareRelations != null)
+            {
+                RelatedEntities = ResolveRelatedEntities(entity, healthcareEntities, healthcareRelations);
+            }
+        }
+
+        private static IReadOnlyDictionary<HealthcareEntity, HealthcareEntityRelationType> ResolveRelatedEntities(HealthcareEntityInternal entity,
+            IEnumerable<HealthcareEntityInternal> healthcareEntities,
+            IEnumerable<HealthcareRelationInternal> healthcareRelations)
+        {
+            Dictionary<HealthcareEntity, HealthcareEntityRelationType> dictionary = new Dictionary<HealthcareEntity, HealthcareEntityRelationType>();
+
+            if (healthcareRelations == null)
+            {
+                return dictionary;
+            }
+
+            foreach (HealthcareRelationInternal relation in healthcareRelations)
+            {
+                int entityIndex = healthcareEntities.ToList().FindIndex(e => e.Text.Equals(entity.Text));
```

Comment:
instead of making `healthcareEntities.ToList` every time in the loop, do it outside the loop

Reply:
actually, I think it will be better to precalculate this before (maybe in the Transform class) similar to how we do the `idToIndexMap`. That way there is a list of entities and their `Ids` and no need to loop through the `healthcareRelations` type for every entity.
Currently complexity is O(n2), with new suggestion it will be O(n)

Reply:
>instead of doing it this way.

@deyaaeldeen do you the current implementation? or my suggestion?

Reply:
I meant to instead of looking for entities by text, just parse the JSON pointer. Here is how I am doing it in JS: https://github.com/Azure/azure-sdk-for-js/blob/a4e9ce5921fdba54a636228d5ddffab7027332b2/sdk/textanalytics/ai-text-analytics/src/analyzeHealthcareEntitiesResult.ts#L212.

Reply:
@suhas92 look at what JS is doing. Either what I am proposing or what JS did will work. both are O(n) and don't rely on the lookup of the entity on the list.
Modify the code to either solution.